test/unit-tests.c

```diff
@@ -1202,11 +1203,157 @@ test_migration_state_transitions(void **state __attribute__ ((unused)))
  * and provide a function to execute before and after each unit test.
  */
 static int
-setup(void **state __attribute__((unused))) {
+setup(void **state __attribute__((unused)))
+{
     unpatch_all();
     return 0;
 }
 
+static struct test_setup_migr_reg_dat {
+    vfu_ctx_t *v;
+    size_t rs; /* migration registers size */
+    size_t ds; /* migration data size */
+    size_t s; /* migration region size*/
+    const vfu_migration_callbacks_t c;
+} migr_reg_data = {
+    .c = {
+        .version = VFU_MIGR_CALLBACKS_VERS,
+        .transition = (void*)0x1,
+        .get_pending_bytes = (void*)0x2,
+        .prepare_data = (void*)0x3,
+        .read_data = (void*)0x4,
+        .write_data = (void*)0x5,
+        .data_written = (void*)0x6
+    }
+};
+
+static int
+setup_test_setup_migration_region(void **state) {
+    struct test_setup_migr_reg_dat *p = &migr_reg_data;
+    p->v = vfu_create_ctx(VFU_TRANS_SOCK, "test", 0, NULL,
+        VFU_DEV_TYPE_PCI);
+    if (p->v == NULL) {
+        return -1;
+    }
+    p->rs = ROUND_UP(sizeof(struct vfio_device_migration_info), sysconf(_SC_PAGE_SIZE));
+    p->ds = sysconf(_SC_PAGE_SIZE);
+    p->s = p->rs + p->ds;
+    *state = p;
+    return setup(state);
+}
+
+static vfu_ctx_t *
+get_vfu_ctx(void **state)
+{
+    return (*((struct test_setup_migr_reg_dat**)(state)))->v;
+}
+
+static int
+teardown_test_setup_migration_retion(void **state)
```

Comment:
nit, "region" ?

Reply:
ok

Reply:
'region' isn't that accurate, since the pointer points to `struct test_setup_migr_reg_dat`, which is 'migration region test data'. Honestly I think it'f fine as 'state' since the cast is right below.

Reply:
It's the "retion" part I was talking about

Reply:
ah yes